plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -39,7 +48,45 @@ export class TeamManager {
         }
     }
 
-    public async updateEventNamesAndProperties(teamId: number, event: string, properties: Properties): Promise<void> {
+    async flushLastSeenAtCache(): Promise<void> {
+        let valuesStatement = ''
+        let params: (string | number)[] = []
+
+        const events = this.eventLastSeenCache
+        this.eventLastSeenCache = new Map()
+
+        for (const event of events) {
+            const [key, value] = event
+            const teamId = key.split('_', 1)[0]
+            const eventName = key.substring(key.indexOf(teamId) + teamId.length + 1)
```

Comment:
This part is confusing and it's not obvious that `split('_', 2)` wouldn't work. I'd either add a comment explaining why `substring` needs to be used, or replace it with something cleaner like:

```
// split('_', 2) eats chunks 3 and up, so we must do this:
const [teamId, ...rest] = key.split('_')
const eventName = rest.join('_')
```

Reply:
The other option is to use as a key: `JSON.stringify([teamId, eventName])`... and read it back with `const [teamId, eventName] = JSON.parse(key)`

Reply:
Like the JSON idea, done!